### PR TITLE
[factorio]: add chart for factorio server

### PR DESCRIPTION
**Pull request by Truxnell**
**Description of the change**

Add a new helm chart for a Factorio game server

**Benefits**

New helm chart for factorio, complete with some QOL to enable users to set all server parameters in yaml over having to exec in to change config maually

**Possible drawbacks**

Maintenance.  New/changed server configs may require chart updates.  That said, the game is released >1.0.0 so devs dont plan changes until expansion next year.

**Applicable issues**

Nil

**Additional information**

Few points on this chart & design logic so you get my thought process:

* Ive been using this for a few weeks (just `helm install . -f values.yaml`), done some fixes and running well.

* Using sable not latest - Stable follows the official stable release & what steam uses.  Latest at times can get into beta territory.

* Created a series of configmaps to allow users to set game server values in the helm chart over expecting people to modify all the config manually.  As these are mounted readonly and the server really doesnt ike that, I have used a initcontainer to mount them in a different location and copied them over.  The configmap is populated with all the defaults that come with the server files.

* I also did a little bit of wierdness with liveness.  Id like there to be a liveness probe and one trick is setting the RCOM port to being main so k8s pings it for liveness.  It works well, with the drawback it spams the server console with pings.

* I have excluded this in the ct-test as its a large image.  Quite happy to have it tested/att a ct-test yaml w/ emptydir mount but doubt we want to spam github with large image pulls.

**Checklist** 
- [X] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Chart `artifacthub.io/changes` changelog annotation has been updated in `Chart.yaml`. See [Artifact Hub documentation](https://artifacthub.io/docs/topics/annotations/helm/#supported-annotations) for more info.
- [X] Variables have been documented in the `values.yaml` file.

### CODE REVIEW

File: charts/stable/factorio/values.yaml [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+  main:
+    enabled: false
+
+# -- Configure persistence settings for the chart under this key.
+# @default -- See values.yaml
+persistence:
+  config:
+    enabled: true
+    mountPath: /factorio
+# -- Configure server settings.  Can be disabled if you wish to configure in the container in the json config files.
```
Comment (bjw-s) [inline charts/stable/factorio/values.yaml:89]:
```suggestion
    mountPath: /factorio

# -- Configure server settings.  Can be disabled if you wish to configure in the container in the json config files.
```
Also maybe add a reference here to somewhere that explains these settings?

Comment (Truxnell) [reply charts/stable/factorio/values.yaml]: Done - pushed another commit to this branch.  Unfortunately not a good place to find all the setting descriptions online however, the official docs assume the var name is enough in some spots :(

**Comment by Truxnell** [comment]
Fixed (with a bit more commentary in the docs).  Honestly I was expecting a bigger review!

**Comment by bjw-s** [comment]
> Honestly I was expecting a bigger review!

This is a pretty specialized chart, so I can't really comment on it _that_ much. These typically take a few point releases to iron out the small things.

**Review by bjw-s — APPROVED** [review]

**Pull request merged**